### PR TITLE
[ML] return zxx for lang_ident_model_1 if no valid text is found for language identification

### LINKED ISSUE #81933: [ML] What to do about lang_ident for empty strings and numbers?
It's been [pointed out](https://discuss.elastic.co/t/default-language-for-language-detection/292470) that the lang_ident inference processor returns Japanese as the language for empty strings and numbers.

For example:

```
{
      "doc" : {
        "_index" : "_index",
        "_id" : "_id",
        "_source" : {
          "contents" : "",
          "_ml" : {
            "lang_ident" : {
              "prediction_score" : 0.7837568024575047,
              "model_id" : "lang_ident_model_1",
              "top_classes" : [
                {
                  "class_name" : "ja",
                  "class_probability" : 0.7837568024575047,
                  "class_score" : 0.7837568024575047
                },
                {
                  "class_name" : "ko",
                  "class_probability" : 0.14699680203424537,
                  "class_score" : 0.14699680203424537
                },
                {
                  "class_name" : "sr",
                  "class_probability" : 0.04528638971813643,
                  "class_score" : 0.04528638971813643
                }
              ],
              "prediction_probability" : 0.7837568024575047,
              "predicted_value" : "ja"
            }
          }
        },
        "_ingest" : {
          "timestamp" : "2021-12-20T14:46:31.19367Z"
        }
      }
```

Since we have nothing to go on in these cases, a configurable default is probably the best we can do.  Alternatively we could treat the absence of any character in any alphabet as an error, and use the failure handler functionality of ingest processors to allow the user to supply the alternative processors to use in this case (which could just be a set processor to apply a default).  Or maybe there's an even better solution.  But we shouldn't just predict Japanese in this situation.

### PR DESCRIPTION
**Pull request by benwtrent**
If no text is available from which to infer the language, we should return a special case instead of defaulting to `ja`. This commit adds a special value `zxx` to the list of possible classifications. `zxx` indicates that no valid text was found and classification could not occur. 

If users want to rely on a "default classification" instead of `zxx`, they can adjust the their inference pipelines to replace classifications of `zxx` with whatever their preferred default is.

closes: https://github.com/elastic/elasticsearch/issues/81933

### CODE REVIEW

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/langident/LangIdentNeuralNetwork.java [hunk 1]
```diff
@@ -161,11 +161,13 @@
         "pa",
         "ku"
     );
+    // The final value for our provided language names indicates if there is no valid text in the string to classify
+    // This value should map back to "zxx"
```
Comment (droberts195) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/langident/LangIdentNeuralNetwork.java:165]:
```suggestion
    // This value should map back to "zxx", which is the code for "no linguistic content" in the BCP 47
    // system - see https://tools.ietf.org/search/bcp47
```

**Review by droberts195 — COMMENTED** [review]
Please could you also add `zxx` to the table in https://github.com/elastic/stack-docs/blob/main/docs/en/stack/ml/nlp/ml-nlp-lang-ident.asciidoc, and also add a tip to that page saying that if you want to apply a default language when there is no linguistic content in a string then you should do it by adding a subsequent processor in the ingest pipeline that replaces `zxx` with your chosen default language.

**Comment by benwtrent** [comment]
@droberts195 please see https://github.com/elastic/stack-docs/pull/1944 for the docs update

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/part-2

**Pull request merged**